bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieNettyServer.java

```diff
@@ -279,6 +280,12 @@ void start() {
     void shutdown() {
         LOG.info("Shutting down BookieNettyServer");
         isRunning.set(false);
+
+        if (!isClosed.compareAndSet(false, true)) {
```

Comment:
IMHO maybe you can move this at the start of the method, so we will not print twice "Shutting down" and doing to isRunning.set